src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -68,13 +124,46 @@ protected RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNul
     }
     FileDocumentManager.getInstance().saveAllDocuments();
     CoverageEnabledConfiguration coverageEnabledConfiguration = CoverageEnabledConfiguration.getOrCreate(runConfiguration);
-    runningState.setCoverageFilePath(coverageEnabledConfiguration.getCoverageFilePath());
+    CoveragePatcher runningStatePatcher = new CoveragePatcher(coverageEnabledConfiguration, runConfiguration);
+    runningState.setPatcher(runningStatePatcher);
+
+    if (runConfiguration.getKind() == GoTestRunConfiguration.Kind.DIRECTORY &&
+        !runningStatePatcher.hasPackageCoverageExecutable()) {
+      promptForPackageCoverageTool(runConfiguration);
+    }
 
     ExecutionResult executionResult = state.execute(environment.getExecutor(), this);
     if (executionResult == null) {
       return null;
     }
+    executionResult.getProcessHandler().addProcessListener(new GoCoverageProcessListener(
+        runConfiguration.getDirectoryPath(),
```

Comment:
Yes, each of items is important.

> The only way to do this with your points considered will be for the IDE

It's not the only way. I believe `package-coverage` is not the only tool for multi-package coverage. The PR can be applied to any of such tool. These packages I found for couple minutes:

https://github.com/go-playground/overalls
https://github.com/haya14busa/goverage
https://github.com/pierrre/gotestcover
https://github.com/axw/gocov

Also, `package-coverage` can be changed as well to allow customize output directory.